src/compositemark/boxplot.ts

```diff
@@ -165,86 +160,193 @@ export function normalizeBoxPlot(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT
         resultOrient = 'vertical';
-      } else if (yDef.aggregate === undefined && xDef.aggregate === BOXPLOT) {
+      } else if (isDiscrete(encoding.y) && isContinuous(encoding.x)) {
         resultOrient = 'horizontal';
-      } else if (xDef.aggregate === BOXPLOT && yDef.aggregate === BOXPLOT) {
-        throw new Error('Both x and y cannot have aggregate');
       } else {
-        if (isBoxPlotDef(mark)) {
-          if (mark && mark.orient) {
-            resultOrient = mark.orient;
-          } else {
-            // default orientation = vertical
+        if (isContinuous(encoding.x) && isContinuous(encoding.y)) {
+          if (encoding.x.aggregate === undefined && encoding.y.aggregate === BOXPLOT) {
             resultOrient = 'vertical';
+          } else if (encoding.y.aggregate === undefined && encoding.x.aggregate === BOXPLOT) {
+            resultOrient = 'horizontal';
+          } else if (encoding.x.aggregate === BOXPLOT && encoding.y.aggregate === BOXPLOT) {
+            throw new Error('Both x and y cannot have aggregate');
+          } else {
+            if (isBoxPlotDef(mark)) {
+              if (mark && mark.orient) {
+                resultOrient = mark.orient;
+              } else {
+                // default orientation = vertical
+                resultOrient = 'vertical';
+              }
+            } else {
+              resultOrient = 'vertical';
+            }
           }
         } else {
-          resultOrient = 'vertical';
+          throw new Error('Both x and y cannot be discrete');
         }
       }
+
+      return resultOrient;
     } else {
-      throw new Error('Both x and y cannot be discrete');
+      throw new Error('Both axes must be valid field definitions');
     }
   }
-
-  return resultOrient;
 }
 
 
-export function box2DParams(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>, orient: Orient) {
+export function boxParams(spec: GenericUnitSpec<Encoding<Field>, BOXPLOT | BoxPlotDef>, orient: Orient) {
   const {mark: mark, encoding: encoding, ...outerSpec} = spec;
 
   let discreteAxisFieldDef: PositionFieldDef<Field>;
   let continuousAxisChannelDef: PositionFieldDef<Field>;
   let discreteAxis;
   let continuousAxis;
 
-  // FIXME: refactor code such that we don't have to do this casting
-  // We can cast here as we already check from outside that both x and y are FieldDef
-  const xDef = encoding.x as FieldDef<Field>;
-  const yDef = encoding.y as FieldDef<Field>;
-
+  const is2D = encoding.x && encoding.y;
 
-  if (orient === 'vertical') {
-    discreteAxis = 'x';
-    continuousAxis = 'y';
-    continuousAxisChannelDef = yDef;
-    discreteAxisFieldDef = xDef;
+  if (!is2D) {
+    if (orient === 'horizontal') {
+      continuousAxis = 'x';
+      if (isFieldDef(encoding.x)) {
+        continuousAxisChannelDef = encoding.x;
+      }
+    } else {
+      continuousAxis = 'y';
+      if (isFieldDef(encoding.y)) {
```

Comment:
Can this really be false?   If it's not a field def, it shouldn't even be a continousAxis from the beginning.